diff --git a/node_modules/dexie/dist/dexie.js b/node_modules/dexie/dist/dexie.js
index 71c9978..609377c 100644
--- a/node_modules/dexie/dist/dexie.js
+++ b/node_modules/dexie/dist/dexie.js
@@ -5147,13 +5147,15 @@
             }
             catch (_a) { }
         });
-        addEventListener('storage', function (ev) {
-            if (ev.key === STORAGE_MUTATED_DOM_EVENT_NAME) {
-                var data = JSON.parse(ev.newValue);
-                if (data)
-                    propagateLocally(data.changedParts);
-            }
-        });
+        if (typeof addEventListener !== 'undefined') {
+            addEventListener('storage', function(ev) {
+                if (ev.key === STORAGE_MUTATED_DOM_EVENT_NAME) {
+                    var data = JSON.parse(ev.newValue);
+                    if (data)
+                        propagateLocally(data.changedParts);
+                }
+            });
+        }
         var swContainer = self.document && navigator.serviceWorker;
         if (swContainer) {
             swContainer.addEventListener('message', propagateMessageLocally);
